### PR TITLE
Add valid_placement? method and test

### PR DESCRIPTION
**Pull request by coreycartercodes**
May need to refactor to to consider out of order input arrays?

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -7,28 +7,95 @@ class Board
...
+      "B3": Cell.new('B3'),
+      "B4": Cell.new('B4'),
+      "C1": Cell.new('C1'),
+      "C2": Cell.new('C2'),
+      "C3": Cell.new('C3'),
+      "C4": Cell.new('C4'),
+      "D1": Cell.new('D1'),
+      "D2": Cell.new('D2'),
+      "D3": Cell.new('D3'),
+      "D4": Cell.new('D4')
```
Comment (coreycartercodes) [inline lib/board.rb:25]: Changed keys to strings as in the interaction pattern

File: lib/board.rb [hunk 2]
```diff
@@ -7,28 +7,95 @@ class Board
...
+#
+# range_row_array = ("A".."D").to_a
+# range_col_array = ("1".."4").to_a
+#
+# p cross_ref_row = range_row_array & row_array
+# p cross_ref_row.sort == row_array
+#
+# p cross_ref_col = range_col_array & col_array
+# p cross_ref_col.sort == col_array
+
```
Comment (coreycartercodes) [inline lib/board.rb:99]: Used as "sandbox" and can be deleted before final commit.

File: lib/board.rb (hunk 1 above)
Comment (carson-jardine) [reply lib/board.rb:25]: Good catch! Guess I had hashes on my mind haha.

File: lib/board.rb (hunk 2 above)
Comment (carson-jardine) [reply lib/board.rb:99]: From what I can tell, this looks great! I'm going to merge and play around a little to see if anything sticks out to me but it looks organized and easy to read/understand.

**Review by carson-jardine — APPROVED** [review]

**Pull request merged**